File: backend/src/main/java/com/lms_erp/employee/service/HolidayService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.HolidayRequest;
import com.lms_erp.employee.dto.HolidayResponse;

import java.util.List;

public interface HolidayService {

    HolidayResponse createHoliday(HolidayRequest request);

    HolidayResponse updateHoliday(Long holidayId,
                                  HolidayRequest request);

    HolidayResponse getHolidayById(Long holidayId);

    List<HolidayResponse> getAllHolidays();

    void deleteHoliday(Long holidayId);
}

File: lms-frontend/src/employee/services/holidayService.js
// holidayService.js

import api from "../../api/axios";

const BASE_URL = "/api/holidays";

const getAllHolidays = () => {

    return api.get(
        BASE_URL
    );
};

const getHolidayById = (holidayId) => {

    return api.get(
        `${BASE_URL}/${holidayId}`
    );
};

const createHoliday = (holidayData) => {

    return api.post(
        BASE_URL,
        holidayData
    );
};

const updateHoliday = (
    holidayId,
    holidayData
) => {

    return api.put(
        `${BASE_URL}/${holidayId}`,
        holidayData
    );
};

const deleteHoliday = (holidayId) => {

    return api.delete(
        `${BASE_URL}/${holidayId}`
    );
};

const holidayService = {

    getAllHolidays,
    getHolidayById,
    createHoliday,
    updateHoliday,
    deleteHoliday
};

export default holidayService;